kafka-connect-redis/src/main/scala/com/datamountaineer/streamreactor/connect/redis/sink/writer/RedisInsertSortedSet.scala

```diff
@@ -0,0 +1,81 @@
+package com.datamountaineer.streamreactor.connect.redis.sink.writer
+
+import com.datamountaineer.streamreactor.connect.redis.sink.config.{RedisKCQLSetting, RedisSinkSettings}
+import com.datamountaineer.streamreactor.connect.rowkeys.StringStructFieldsStringKeyBuilder
+import org.apache.kafka.connect.sink.SinkRecord
+import scala.collection.JavaConversions._
+import scala.util.Try
+import java.util.Date
+
+/**
+  * A generic Redis `writer` that can store data into 1 Sorted Set / KCQL
+  *
+  * Requires KCQL syntax:   INSERT .. SELECT .. STOREAS SS
+  *
+  * If a field <timestamp> exists it will automatically be used to `score` each value inside the (sorted) set
+  * Otherwise you would need to explicitly define how the values will be scored
+  *
+  * Examples:
+  *
+  * INSERT INTO cpu_stats SELECT * from cpuTopic STOREAS SS
+  * INSERT INTO cpu_stats_SS SELECT * from cpuTopic STOREAS SS (score=ts)
+  */
+class RedisInsertSortedSet(sinkSettings: RedisSinkSettings) extends RedisWriter {
+
+  apply(sinkSettings)
+
+  val configs = sinkSettings.allKCQLSettings.map(_.kcqlConfig)
+  configs.foreach { c =>
```

Comment:
My view is this checks should have been done in the source initialization. we want to fail fast. This classes could be private and this checks should be omitted